kodein-type/src/jvmMain/kotlin/org/kodein/type/JVMGenericArrayTypeToken.kt

```diff
@@ -2,76 +2,57 @@ package org.kodein.type
-//            is ParameterizedType -> trueType.rawType as Class<*>
-//            else -> null
-//        }
-//
-//    override fun getGenericParameters(): Array<TypeToken<*>> {
-//        val type = wrappedType as? ParameterizedType
-//                ?: return rawType?.typeParameters?.map { typeToken(it.bounds[0]) } ?.toTypedArray() ?: emptyArray()
-//        return type.actualTypeArguments.map {
-//            if (it is WildcardType)
-//                typeToken(it.upperBounds[0])
-//            else
-//                typeToken(it)
-//        }.toTypedArray()
-//    }
-//
-//    override val jvmType: Type = wrappedType ?: run {
-//        // TypeReference cannot create WildcardTypes nor TypeVariables
-//        when {
-//            KodeinWrappedType.IsNeeded.forParameterizedType.not() && KodeinWrappedType.IsNeeded.forGenericArray.not() -> trueType
-//            trueType is Class<*> -> trueType
-//            KodeinWrappedType.IsNeeded.forParameterizedType && trueType is ParameterizedType -> KodeinWrappedType(trueType)
-//            KodeinWrappedType.IsNeeded.forGenericArray && trueType is GenericArrayType -> KodeinWrappedType(trueType)
-//            else -> trueType
-//        }.also { wrappedType = it }
-//    }
-//
-//    @Suppress("UNCHECKED_CAST")
-//    override fun getRaw(): TypeToken<T>? {
-//        return rawType?.let { JVMClassTypeToken(it as Class<T>) }
-//    }
-//
-//    override fun isGeneric() = true
-//
-//    @Suppress("UNCHECKED_CAST")
-//    override fun isWildcard(): Boolean {
-//        val realType = trueType as? ParameterizedType ?: return false
-//
-//        var hasWildCard = false
-//        var hasSpecific = false
-//
-//        val cls = realType.rawType as Class<*>
-//        cls.typeParameters.forEachIndexed { i, variable ->
-//            val argument = realType.actualTypeArguments[i]
-//
-//            if (argument is WildcardType && variable.bounds.any { it in argument.upperBounds })
-//                hasWildCard = true
-//            else
-//                hasSpecific = true
-//        }
-//
-//        return hasWildCard && !hasSpecific
-//    }
-//
-//    @Suppress("UNCHECKED_CAST")
-//    override fun getSuper(): List<TypeToken<*>> {
-//        val extends = when (val jvmType = jvmType) {
-//            is ParameterizedType -> (jvmType.rawType as Class<T>).superTypeToken()
-//            else -> null
-//        }
-//        val implements = rawType?.genericInterfaces?.map { typeToken(it) } ?: emptyList()
-//        return (extends?.let { listOf(it) } ?: emptyList()) + implements
-//    }
-//}
\ No newline at end of file
+internal class JVMGenericArrayTypeToken<T>(private val trueType: GenericArrayType) : JVMAbstractTypeToken<T>() {
+    override val jvmType: Type = trueType.genericComponentType
+
+    private val rawType: Class<*>?
+        get() = when (jvmType) {
+            is Class<*> -> jvmType
+            is ParameterizedType -> jvmType.rawClass
+            else -> null
+        }
```

Comment:
Same as above.

Should be:

```suggestion
    private val rawType: Class<*>?
        get() {
            val componentClass = typeToken(trueType.genericComponentType).getRaw().jvmType
            return java.lang.reflect.Array.newInstance(componentClass, 0).javaClass
        }
```